src/app/cli/cli.ts

```diff
@@ -3,21 +3,57 @@ import chalk from 'chalk';
 import { logger } from '@core/index.js';
 import { CLISubscriber } from './cli-subscriber.js';
 import { SaikiAgent } from '@core/index.js';
+import { parseInput } from './command-parser.js';
+import { executeCommand } from './commands.js';
 
-const validLogLevels = ['error', 'warn', 'info', 'http', 'verbose', 'debug', 'silly'];
-const HELP_MESSAGE = `Available commands:
-exit/quit - Exit the CLI
-clear - Clear conversation history
-help - Show this help message
-currentloglevel - Show current logging level
-${validLogLevels.join('|')} - Set logging level directly
-`;
+/**
+ * Find and load the most recent session based on lastActivity.
+ * This provides better UX than always loading the "default" session.
+ */
+async function loadMostRecentSession(agent: SaikiAgent): Promise<void> {
+    try {
+        const sessionIds = await agent.listSessions();
+
+        if (sessionIds.length === 0) {
+            // No sessions exist, let agent create default
+            logger.debug('No existing sessions found, will use default session');
+            return;
+        }
+
+        // Find the session with the most recent activity
+        let mostRecentSession = sessionIds[0];
+        let mostRecentActivity = 0;
+
+        for (const sessionId of sessionIds) {
+            const metadata = await agent.getSessionMetadata(sessionId);
+            if (metadata && metadata.lastActivity > mostRecentActivity) {
+                mostRecentActivity = metadata.lastActivity;
+                mostRecentSession = sessionId;
+            }
+        }
+
+        // Load the most recent session if it's not already current
+        const currentSessionId = agent.getCurrentSessionId();
+        if (mostRecentSession !== currentSessionId) {
+            await agent.loadSession(mostRecentSession);
+            logger.debug(`Loaded most recent session: ${mostRecentSession}`);
```

Comment:
we can make this logger.info and with colour so that it shows users which session is loaded